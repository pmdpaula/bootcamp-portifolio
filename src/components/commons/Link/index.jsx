/* eslint-disable import/prefer-default-export */
import get from 'lodash/get';
import NextLink from 'next/link';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledLink = styled.a`
  color: inherit;
  ${({ theme, color }) =>
    color
      ? `color: ${get(theme, `colors.${color}.color`)}`
      : 'color: inherit;'};
  text-decoration: none;
  opacity: 1;
  transition: opacity ${({ theme }) => theme.transition};
  &:hover,
  &:focus {
    opacity: 0.5;
  }
`;

const Link = ({ href, children, ...props }) => (
  <NextLink href={href} passHref>
    <StyledLink {...props}>{children}</StyledLink>
  </NextLink>
);

export default Link;

Link.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
