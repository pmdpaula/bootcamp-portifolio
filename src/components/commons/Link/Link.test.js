import { render, screen } from '../../../infra/test/testUtils';
import Link from './index';

describe('<Button />', () => {
  // const { colors } = useContext(ThemeContext);
  test('renders component', () => {
    render(<Link href="/">Link</Link>);

    // screen.debug();
    // expect(true).toBe(true);

    const button = screen.getByRole('link', { name: /link/i });
    expect(button).toMatchSnapshot();
  });
});
