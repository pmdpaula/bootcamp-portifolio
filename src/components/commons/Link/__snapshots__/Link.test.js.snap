// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders component 1`] = `
.c0 {
  color: inherit;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  opacity: 1;
  -webkit-transition: opacity;
  transition: opacity;
}

.c0:hover,
.c0:focus {
  opacity: 0.5;
}

<a
  class="c0"
  href="/"
>
  Link
</a>
`;
